src/js/libcs/Evaluator.js

```diff
@@ -151,3 +154,29 @@ function analyse(code) {
         namespace.create(name);
     return res;
 }
+
+var callStack = [];
+
+function labelCode(code, label) {
+    function run() {
+        return evaluate(code);
+    }
+    return {
+        ctype: "infix",
+        args: [],
+        impl: function() {
+            callStack = [{
+                oper: label
+            }];
+            var res = evaluate(code);
+            callStack = [];
+            return res;
+        }
+    };
+}
+
+function printStackTrace(msg) {
+    csconsole.err(msg + "\n" + callStack.map(function(frame) {
+        return "  at " + frame.oper;
```

Comment:
It would be really nice, if line numbers were also printed. But this would be a quite invasive surgery for the `analyse` function.